processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -625,27 +616,16 @@ public Metadata apply(IndexableAdapter input)
     }
 
     Closer closer = Closer.create();
-    final Interval dataInterval;
-    final File v8OutDir = new File(outDir, "v8-tmp");
-    v8OutDir.mkdirs();
-    closer.register(new Closeable()
-    {
-      @Override
-      public void close() throws IOException
-      {
-        FileUtils.deleteDirectory(v8OutDir);
-      }
-    });
-    final IOPeon ioPeon = new TmpFileIOPeon();
-    closer.register(new Closeable()
-    {
-      @Override
-      public void close() throws IOException
-      {
-        ioPeon.close();
-      }
-    });
     try {
+      final Interval dataInterval;
+      final File v8OutDir = new File(outDir, "v8-tmp");
+      FileUtils.forceMkdir(v8OutDir);
+      registerDeleteDirectory(closer, v8OutDir);
+      File tmpPeonFilesDir = new File(v8OutDir, "tmpPeonFiles");
+      FileUtils.forceMkdir(tmpPeonFilesDir);
+      registerDeleteDirectory(closer, tmpPeonFilesDir);
```

Comment:
It's harmless and makes the code a little more "self-contained", if `tmpPeonFilesDir` if moved somewhere else (out of `v8OutDir`), so I would keep these calls